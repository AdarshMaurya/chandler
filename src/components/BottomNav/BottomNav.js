import React from 'react';
import PropTypes from 'prop-types';
import {View, Text, TouchableOpacity} from 'react-native';
import {hitSlop} from '../../theme';
import {styles} from './BottomNav.styles';

export const BottomNav = ({onPress, title, linkTitle}) => (
  <View style={styles.bottomNav}>
    <Text>{title} </Text>
    <TouchableOpacity hitSlop={hitSlop} onPress={onPress}>
      <Text style={styles.bottomLinkText}>{linkTitle}</Text>
    </TouchableOpacity>
  </View>
);

BottomNav.propTypes = {
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  linkTitle: PropTypes.string.isRequired,
};
