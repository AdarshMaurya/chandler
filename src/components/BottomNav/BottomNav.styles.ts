import {PixelRatio, StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  bottomNav: {
    paddingVertical: 15,
    borderTopWidth: 1 / PixelRatio.get(),
    borderColor: '#E2E4E5',
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  bottomLinkText: {
    color: '#1361cb',
  },
});
