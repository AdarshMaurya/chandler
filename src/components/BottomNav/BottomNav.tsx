import React, {PureComponent} from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import {hitSlop} from '../../theme';
import {styles} from './BottomNav.styles';

interface IBottomNavProps {
  onPress: (...args) => any;
  title: string;
  linkTitle: string;
}

export class BottomNav extends PureComponent<IBottomNavProps> {
  public render() {
    const {onPress, title, linkTitle} = this.props;

    return (
      <View style={styles.bottomNav}>
        <Text>{title} </Text>
        <TouchableOpacity hitSlop={hitSlop} onPress={onPress}>
          <Text style={styles.bottomLinkText}>{linkTitle}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
