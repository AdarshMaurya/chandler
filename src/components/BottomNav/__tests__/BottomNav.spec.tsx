import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {BottomNav} from '../BottomNav';

describe('Components', () => {
  describe('BottomNav', () => {
    it('should renders correctly', () => {
      const onPress = jest.fn();
      expect(
        toJson(
          shallow(
            <BottomNav
              onPress={onPress}
              title="Bottom Nav title"
              linkTitle="Link title"
            />
          )
        )
      ).toMatchSnapshot();
    });

    it('should call onPress when taping the icon', () => {
      const onPress = jest.fn();
      const button = shallow(
        <BottomNav onPress={onPress} title="a" linkTitle="b" />
      );
      button.find('TouchableOpacity').prop('onPress')();
      expect(onPress).toHaveBeenCalledTimes(1);
      expect(onPress).toHaveBeenCalledWith();
    });
  });
});
