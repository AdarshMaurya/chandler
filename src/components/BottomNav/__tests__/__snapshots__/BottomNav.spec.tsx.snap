// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components BottomNav should renders correctly 1`] = `
<Component
  style={
    Object {
      "borderColor": "#E2E4E5",
      "borderTopWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingVertical": 15,
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Bottom Nav title
     
  </Text>
  <TouchableOpacity
    activeOpacity={0.2}
    hitSlop={
      Object {
        "bottom": 15,
        "left": 15,
        "right": 15,
        "top": 15,
      }
    }
    onPress={[MockFunction]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1361cb",
        }
      }
    >
      Link title
    </Text>
  </TouchableOpacity>
</Component>
`;
